src/resolve_reports

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+import csv
+from os import environ
+
+from bs4 import BeautifulSoup
+
+from logic import (
+    determine_timestamp,
+    get_options,
+    get_date_from_filename,
+    uri_schema_doi,
+)
+from utils import doi_to_id, set_token_auth
+
+
+XREF_DOI_PREFIX = environ['XREF_DOI_PREFIX']
+
+
+ARGS = [
+    {
+        'val': '--measure',
+        'dest': 'measure',
+        'action': 'store',
+        'default': None,
+        'help': 'Measure URI, e.g. '
+                'https://metrics.operas-eu.org/citations/crossref/v1',
+        'required': True,
+    }, {
+        'val': '--add-headers',
+        'dest': 'add_headers',
+        'action': 'store_true',
+        'default': [],
+        'help': 'Whether to print report headers',
+        'required': False,
+    }, {
+        'val': '--in_file',
+        'dest': 'in_file',
+        'action': 'store',
+        'help': 'File to read Crossref citation data from (XML format).',
+        'required': True,
+    }, {
+        'val': '--out_file',
+        'dest': 'out_file',
+        'action': 'store',
+        'help': 'CSV file to save output to (CSV format).',
+        'required': True,
+    }
+]
+
+HEADER = (
+    'measure_id',
+    'timestamp',
+    'work_uri',
+    'country_uri',
+    'event_uri',
+    'value',
+)
+
+
+def get_crossref_citations(cached_file):
+    """Extract raw data from Crossref citation XML for correct DOIs only.
+
+    Args:
+        cached_file (str): Name of XML file containing data from Crossref.
+
+    Returns:
+        dict: Each DOI that is cited and the citation entries for that DOI.
+    """
+
+    with open(cached_file, 'r') as f:
+        xml_data = BeautifulSoup(f.read(), 'xml')
```

Comment:
Are we supposed to use `lxml` here? If so, you're not. If you are, maybe we should remove lxml from the requirements?

If you want to use `lxml`

```xml_data = BeautifulSoup(f.read(), 'lxml')```

Reply:
Sorry, I thought I saw a requirements.txt file before that included `lxml`. Can't find it now, so must have imagined it.

Reply:
We don't strictly need to use `lxml` here. It was only added to requirements as a fix to this error:
`bs4.FeatureNotFound: Couldn't find a tree builder with the features you requested: xml. Do you need to install a parser library?`

Reply:
Well why not use it directly then? If we go to the trouble of installing it, especially given the length of time it takes to compile, we might as well use it???